### ISSUE TITLE
Route matching works not as expected

### ISSUE THREAD

**Issue by abriginets** [storyofams/next-api-decorators#170]
**Describe the bug**
Route matching doesn't work as intended. Here's my filetree:

![image](https://user-images.githubusercontent.com/17240507/122467678-7ac16880-cfc3-11eb-85c4-4c753ca047ff.png)

My controller code:
```ts
import { createHandler, Get, Param } from '@storyofams/next-api-decorators';

import { getLatestPrices } from '../../../../src/server/clients/travelpayouts';

class LatestPrices {
  @Get()
  async getCheapestPrices() {
    return await getLatestPrices({});
  }

  @Get(':origin')
  async getCheapestPricesFromOrigin(
    @Param('origin') origin: string,
  ) {
    return await getLatestPrices({ origin });
  }
}

export default createHandler(LatestPrices);
```

I'm hitting `http://localhost:3000/api/prices/latest` and it returns `{"statusCode":404,"message":"Cannot GET /api/prices/latest","error":"Not Found"}`
But if I change the controller to look like following - it works:
```ts
@Get('/latest')
async getCheapestPrices() {
  return await getLatestPrices({});
}

@Get('/latest/:origin')
async getCheapestPricesFromOrigin(
  @Param('origin') origin: string,
) {
  console.log(origin);

  return await getLatestPrices({ origin });
}
```

It doesn't really makes sense to add the `/latest` prefix since I'm already under `/pages/api/prices/latest`. And it's something that is not documented. I'd suggest changing the behavior to not be needing to prepend the `latest` to all the endpoints, but that seems to be a breaking change and will require you to bump version to 2.x.x

**Comment by leeuwis** [comment]
Hi @abriginets, thanks for flagging the issue.

One thing I noticed in the first code example, a `/` should be added to `/:origin`. However, with this change I'm still experiencing the same unexpected behaviour as you are describing. I'm using the below code (added in the `examples/route-matching` project) to replicate the issue:

```ts
// pages/api/prices/latest/[[...params]].ts
import { createHandler, Get, Param } from '@storyofams/next-api-decorators';

class LatestPrices {
  @Get()
  async getCheapestPrices() {
    return 'TEST';
  }

  @Get('/:origin')
  async getCheapestPricesFromOrigin(@Param('origin') origin: string) {
    return `TEST ${origin}`;
  }
}

export default createHandler(LatestPrices);
```

When moving the exact same file on level lower to `pages/api/prices/[[...params]].ts`, the routing matching does work as expected. So the issue seems to be related to the nesting of the folders.

Will look into the issue, @ggurkal probably good if you take a look as well.

**Comment by abriginets** [comment]
> One thing I noticed in the first code example, a / should be added to /:origin
I mentioned that too, but that's okay since it's documented and I noticed that only after I created the issue.

**Comment by abriginets** [comment]
@ggurkal @leeuwis now it works fine for `[[...params]]` stuff, but doesn't work for simple routes. Like if I have a route handler under `/pages/api/location/index.ts` I have to specify the route path with `@Get('/location')`

**Comment by ggurkal** [comment]
Thanks @abriginets 

I also spotted another bug related to this. Working on a fix.

**Comment by ggurkal** [comment]
Hey @abriginets, just released a fix (v1.5.4).
